### PR TITLE
feat(ToggleSwitch): Add small size

### LINKED ISSUE #1091: Smaller size for toggle switch
Hey guys, 
we need a smaller size toggle switch as this one doesn't look good/proportional with base text. It's not good either in smaller panels.

image

### PR DESCRIPTION
**Pull request by FlyersPh9**
## Changes

- Swap necessary SASS variables for CSS variables.
- Introduce `data-iui-size="small"` and bump CSS variable values down by 1.
- Hide icon when small size is active.
- Swap hard coded margin values with CSS variables in demo page.

Screenshot 2023-02-23 at 9 49 07 AM

- Added `switchSize` prop to ToggleSwitch type in React, which changes `data-iui-size` depending on whether it is `small` or `default`. Default is `default` if nothing is specified.
- Added "Small" under Toggle Switch in storybook to demonstrate the smaller toggle switch based on the passed in `switchSize` prop.

image
image

~~Requires React portion,~~ closes #1091.

## Testing

- BackstopJS

## Docs

N/A

### CODE REVIEW

File: .changeset/eleven-mangos-mix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-css': minor
```
Comment (mayank99) [inline .changeset/eleven-mangos-mix.md:2]:
```suggestion
'@itwin/itwinui-css': minor
'@itwin/itwinui-react': minor
```

File: apps/storybook/src/ToggleSwitch.stories.tsx [hunk 2]
```diff
@@ -23,6 +23,19 @@ export const Basic: Story<ToggleSwitchProps> = (args) => {
   return <ToggleSwitch defaultChecked {...args} />;
 };
 
+export const Small: Story<ToggleSwitchProps> = (args) => {
+  const { switchSize = 'small', ...rest } = args;
+  return (
+    <div>
```
Comment (mayank99) [inline apps/storybook/src/ToggleSwitch.stories.tsx:29]: shouldn't need `div`

File: .changeset/eleven-mangos-mix.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Add small size toggle switch option.
```
Comment (mayank99) [inline .changeset/eleven-mangos-mix.md:5]: you can add a tiny code snippet here showing how to use it

```tsx
<ToggleSwitch switchSize='small' />
```

File: packages/itwinui-react/src/core/ToggleSwitch/ToggleSwitch.tsx [hunk 4]
```diff
@@ -26,6 +26,11 @@ export type ToggleSwitchProps = {
    * Icon inside the toggle switch. Shown only when toggle is checked.
    */
   icon?: JSX.Element;
+  /**
+   * Size of the toggle switch.
+   *  @default 'default'
+   */
+  switchSize?: 'default' | 'small';
```
Comment (mayank99) [inline packages/itwinui-react/src/core/ToggleSwitch/ToggleSwitch.tsx:33]: The `switchSize` name is a bit weird. Maybe we should use `size`? Only concern there is potential clash with the [native `size`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/size) attribute, but that is not applicable to checkbox inputs, so we should be fine.

File: .changeset/eleven-mangos-mix.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+'@itwin/itwinui-css': minor
+'@itwin/itwinui-react': minor
+---
+
+Add small size toggle switch option using `size` prop as follow: `<ToggleSwitch size='small' />`
```
Comment (mayank99) [inline .changeset/eleven-mangos-mix.md:6]:
```suggestion
Add small size toggle switch option using `size` prop as follows: `<ToggleSwitch size='small' />`
```

**Review by mayank99 — APPROVED** [review]
Looks good 🚀

File: .changeset/eleven-mangos-mix.md (hunk 5 above)
Comment (LostABike) [reply .changeset/eleven-mangos-mix.md:6]: Hold on, changing variable name to `size` was breaking my code which was why I named it `switchSize`. However it is not anymore but it is breaking the unit test. Do you happen to know why?

`
  ● Test suite failed to run

    src/core/ToggleSwitch/ToggleSwitch.test.tsx:35:46 - error TS2322: Type 'string' is not assignable to type 'undefined'.

    35   const { container } = render(<ToggleSwitch size='small' />);
                                                    ~~~~

      src/core/ToggleSwitch/ToggleSwitch.tsx:33:3
        33   size?: 'default' | 'small';
             ~~~~
        The expected type comes from property 'size' which is declared here on type 'IntrinsicAttributes & { label?: ReactNode; labelPosition?: "left" | "right" | undefined; setFocus?: boolean | undefined; icon?: Element | undefined; size?: "small" | ... 1 more ... | undefined; } & Omit<...> & RefAttributes<...>'
        `

File: packages/itwinui-react/src/core/ToggleSwitch/ToggleSwitch.tsx [hunk 6]
```diff
@@ -26,6 +26,11 @@ export type ToggleSwitchProps = {
    * Icon inside the toggle switch. Shown only when toggle is checked.
    */
   icon?: JSX.Element;
+  /**
+   * Size of the toggle switch.
+   *  @default 'default'
+   */
+  size?: 'default' | 'small';
 } & Omit<React.InputHTMLAttributes<HTMLInputElement>, 'type'>;
```
Comment (mayank99) [inline packages/itwinui-react/src/core/ToggleSwitch/ToggleSwitch.tsx:34]: This should fix your test failure mentioned in https://github.com/iTwin/iTwinUI/pull/1092#discussion_r1123425633
```suggestion
} & Omit<React.InputHTMLAttributes<HTMLInputElement>, 'type' | 'size'>;
```

Comment (LostABike) [reply packages/itwinui-react/src/core/ToggleSwitch/ToggleSwitch.tsx]: Oh interesting! Thanks 😸

**Review by gretanausedaite — APPROVED** [review]

**Pull request merged**